airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/record.py

```diff
@@ -0,0 +1,119 @@
+        component = record.get("__typename")
+        record.pop("__typename")
+        # add compponent to it's placeholder in the components list
+        self.buffer[-1]["record_components"][component].append(record)
+
+    def component_prepare(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        if self.components:
+            record["record_components"] = {}
+            for component in self.components:
+                record["record_components"][component] = []
+        return record
+
+    def buffer_flush(self) -> Iterable[Mapping[str, Any]]:
+        if len(self.buffer) > 0:
+            for record in self.buffer:
+                # resolve id from `str` to `int`
+                record = self.record_resolve_id(record)
+                # process record components
+                yield from self.record_process_components(record)
+            # clean the buffer
+            self.buffer.clear()
+
+    def record_compose(self, record: Mapping[str, Any]) -> Optional[Iterable[Mapping[str, Any]]]:
+        """
+        Step 1: register the new record by it's `__typename`
+        Step 2: check for `components` by their `__typename` and add to the placeholder
+        Step 3: repeat until the `<END_OF_FILE>`.
+        """
+
+        if self.check_type(record, self.composition.get("new_record")):
+            # emit from previous iteration, if present
+            yield from self.buffer_flush()
+            # register the record
+            self.record_new(record)
+        # components check
+        elif self.check_type(record, self.components):
+            self.record_new_component(record)
+
+    def process_line(self, jsonl_file: TextIOWrapper) -> Iterable[Mapping[str, Any]]:
+        # process the json lines
+        for line in jsonl_file:
+            # we exit from the loop when receive <end_of_file> (file ends)
+            if line == END_OF_FILE:
+                break
+            elif line != "":
+                yield from self.record_compose(loads(line))
+
+        # emit what's left in the buffer, typically last record
+        yield from self.buffer_flush()
+
+    def record_resolve_id(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        The ids are fetched in the format of: " gid://shopify/Order/<Id> "
+        Input:
+            { "Id": "gid://shopify/Order/19435458986123"}
+        We need to extract the actual id from the string instead.
+        Output:
+            { "id": 19435458986123, "admin_graphql_api_id": "gid://shopify/Order/19435458986123"}
+        """
+        # save the actual api id to the `admin_graphql_api_id`
+        # while resolving the `id` in `record_resolve_id`,
+        # we re-assign the original id like `"gid://shopify/Order/19435458986123"`,
+        # into `admin_graphql_api_id` have the ability to identify the record oigin correctly in subsequent actions.
+        id = record.get("id")
+        if id and isinstance(id, str):
```

Comment:
what other types of id are expected? what should we do if we encounter them? I naively don't think ignoring them is the right behavior

Reply:
Other ones could be `int` already. Added the check to return `as is` if `int` is faced.